server/parameters/ConfigParser.java

```diff
@@ -0,0 +1,472 @@
+import static com.vaticle.typedb.core.server.parameters.util.YamlParser.Value.Primitive.BYTES_SIZE;
+import static com.vaticle.typedb.core.server.parameters.util.YamlParser.Value.Primitive.INET_SOCKET_ADDRESS;
+import static com.vaticle.typedb.core.server.parameters.util.YamlParser.Value.Primitive.LIST_STRING;
+import static com.vaticle.typedb.core.server.parameters.util.YamlParser.Value.Primitive.PATH;
+import static com.vaticle.typedb.core.server.parameters.util.YamlParser.Value.Primitive.STRING;
+import static com.vaticle.typedb.core.server.parameters.util.YamlParser.dynamic;
+import static com.vaticle.typedb.core.server.parameters.util.YamlParser.predefined;
+import static com.vaticle.typedb.core.server.parameters.util.YamlParser.restricted;
+
+public class ConfigParser extends YamlParser.Value.Compound<Config> {
+
+    private static final Predefined<Config.Server> server = predefined(Server.name, Server.description, new Server());
+    private static final Predefined<Config.Storage> storage = predefined(Storage.name, Storage.description, new Storage());
+    private static final Predefined<Config.Log> log = predefined(Log.name, Log.description, new Log());
+    private static final Predefined<Config.VaticleFactory> vaticleFactory =
+            predefined(VaticleFactory.name, VaticleFactory.description, new VaticleFactory());
+    private static final Set<Predefined<?>> parsers = set(server, storage, log, vaticleFactory);
+
+    @Override
+    public Config parse(Yaml yaml, String path) {
+        if (yaml.isMap()) {
+            validatedRecognisedParsers(parsers, yaml.asMap().keys(), "");
+            return new Config(server.parse(yaml.asMap(), ""), storage.parse(yaml.asMap(), ""),
+                    log.parse(yaml.asMap(), ""), vaticleFactory.parse(yaml.asMap(), "")
+            );
+        } else throw TypeDBException.of(CONFIG_YAML_MUST_BE_MAP, path);
+    }
+
+    @Override
+    public List<HelpEntry> helpEntries(String path) {
+        return list(server.helpEntry(path), storage.helpEntry(path), log.helpEntry(path), vaticleFactory.helpEntry(path));
+    }
+
+    private static Path configPathAbsolute(Path path) {
+        if (path.isAbsolute()) return path;
+        else return Util.getTypedbDir().resolve(path);
+    }
+
+    private static class Server extends Compound<Config.Server> {
+
+        private static final String name = "server";
+        private static final String description = "Server and networking configuration.";
+
+        private static final Predefined<InetSocketAddress> address =
+                predefined("address", "Address to listen for GRPC clients on.", INET_SOCKET_ADDRESS);
+        private static final Set<Predefined<?>> parsers = set(address);
+
+        @Override
+        public Config.Server parse(Yaml yaml, String path) {
+            if (yaml.isMap()) {
+                validatedRecognisedParsers(parsers, yaml.asMap().keys(), path);
```

Comment:
i think this should be called `validatePredefinedParsers` now

Reply:
Ah much better. Fixed in 443186cb5e02eba13fd86437dd8e39d299a5e64c.